pallets/parachain-staking/src/migrations.rs

```diff
@@ -812,3 +814,66 @@ impl<T: Config> OnRuntimeUpgrade for PurgeStaleStorage<T> {
 		Ok(())
 	}
 }
+
+/// Migration to convert currency reserves to locks
+pub struct ConvertReservesToLocks<T>(PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for ConvertReservesToLocks<T> {
+	fn on_runtime_upgrade() -> Weight {
+		use crate::COLLATOR_LOCK_IDENTIFIER;
+		use frame_support::traits::{LockableCurrency, ReservableCurrency, WithdrawReasons};
+
+		log::info!(target: "ConvertReservesToLocks", "starting on_runtime_upgrade()...");
+
+		let (mut num_reads, mut num_writes) = (0, 0);
+		let (mut num_delegators, mut num_collators) = (0, 0);
+
+		for (mut account, mut delegator_state) in <DelegatorState<T>>::iter() {
+			let reserved = delegator_state.total;
+			let remaining = T::Currency::unreserve(&account, reserved);
+			assert_eq!(remaining, 0u32.into());
+
+			delegator_state.adjust_bond_lock::<T>(None);
+
+			num_delegators += 1;
+		}
+
+		log::info!(target: "ConvertReservesToLocks", "migrated {} delegators", num_delegators);
+
+		for (mut account, mut candidate_info) in <CandidateInfo<T>>::iter() {
+			let reserved = candidate_info.bond;
+			let remaining = T::Currency::unreserve(&account, reserved);
+			assert_eq!(remaining, 0u32.into());
+
+			T::Currency::set_lock(
+				COLLATOR_LOCK_IDENTIFIER,
+				&account,
+				reserved,
+				WithdrawReasons::all(),
+			);
+
+			num_collators += 1;
+		}
+
+		log::info!(target: "ConvertReservesToLocks", "migrated {} collators", num_collators);
+
+		// TODO: properly tally
+		T::DbWeight::get().reads(num_reads) + T::DbWeight::get().writes(num_writes)
```

Comment:
```suggestion
		T::DbWeight::get().reads_writes(num_reads, num_writes)
```

Reply:
Ah, good suggestion :) I know this fn exists now.

The migration is going away, though...